### ISSUE TITLE
Raise and catch a more specific error when failing to parse JSON body params

### ISSUE THREAD

**Issue by paulcsmith** [luckyframework/lucky_cli#368]
https://github.com/luckyframework/lucky_cli/blob/3a70132e0e328afde59ff6e878a5c2e62a8f12d3/src/web_app_skeleton/src/actions/errors/show.cr.ecr#L2

Otherwise it will incorrectly catch JSON parsing from other parts of the app. We want to show those errors and print those stacktraces

Also the JSON params parsing should probably still show the debug page in development as it is still useful

**Comment by mwlang** [comment]
If app is in development mode, then just render the detail error page

Replacing: 
```crystal
render_error_page status: 500
```
with
```crystal
render_detailed_exception_page(error)
```
surfaced the JSON parsing error whereas simply rendering 500 status page in development mode just leads to challenging debug situation.

**Comment by paulcsmith** [comment]
Closed by #392
